app/service/file_svc.py

```diff
@@ -169,12 +170,27 @@ def _read(self, filename):
         :param ldflags: A string of ldflags to use when building the go executable
         :param cflags: A string of CFLAGS to pass to the go compiler
         :param buildmode: GO compiler buildmode flag
+        :param build_dir: The path to build should take place in
         :return:
         """
-        os.system(
-            'cd %s && %s %s go build %s -o %s -ldflags=\'%s\' %s' % (build_dir, _go_vars(arch, platform), cflags,
-                                                                     buildmode, output, ldflags, src_fle)
-        )
+        env = copy.copy(os.environ)
+        env['GOARCH'] = arch
+        env['PLATFORM'] = platform
+        if cflags:
+            for cflag in cflags.split(' '):
+                name, value = cflag.split('=')
+                env[name] = value
+
+        build_mode_args = ['-buildmode', buildmode] if buildmode else []
+
+        process = await asyncio.subprocess.create_subprocess_exec('go', 'build', *build_mode_args, '-o', output,
+                                                                  '-ldflags', ldflags, src_fle,
```

Comment:
One of the major issue we had was ensuring the nested build arguments were getting passed through correctly to the local compiler and linker (for shared libraries) - have you tested this with a DLL sandcat? Easiest way to test is use the `stowaway` adversary with Defender turned off. if that works then the DLL should be properly configured (ASLR/DEP enabled).

Reply:
standby while I boot my windows vm

Reply:
the changes in a8b73b7 builds the shared.go correctly -- at least for my test environment.  

I did notice one time running stoaway that the injection link timed out -- might be something to keep our eye on since reflectivepeinjection seems to take a bit of time to run.